### PR TITLE
[4.5]BL-7081 Fix Nepali word count

### PR DESCRIPTION
**Pull request by gmartin7**
* existing regexp was too greedy about splitting
   words on certain Control characters

---
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/3128)

### CODE REVIEW

**Comment by andrew-polk** [comment]
___
*[src/BloomBrowserUI/bookEdit/toolbox/readers/libSynphony/synphony_lib.js, line 450 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/3128#-LcgZaZoA9lKx9SrsNiA:-LcgahU52jYidmfA2AjD:b90vhzh) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/bf5c10123178b8aa8d0a4a3d376120c4448b7bef/src/BloomBrowserUI/bookEdit/toolbox/readers/libSynphony/synphony_lib.js#L450)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

How did you come up with this list? Can you link to a reference?

Should any more tests be added for the additional markers?
Do we already have one or more tests for zero-width space and/or zero-width non-joiner?
</blockquote></details>

I guess your new test covers zero-width non-joiner, so that's ok.

File: src/BloomBrowserUI/bookEdit/toolbox/readers/libSynphony/synphony_lib.js [hunk 1]
```diff
@@ -444,9 +444,24 @@ LibSynphony.prototype.getWordsFromHtmlString = function(textHTML, letters) {
...
+    // Split into words using Separator and some Control characters
+    // e.g. ZERO WIDTH SPACE is a Control charactor.  See http://issues.bloomlibrary.org/youtrack/issue/BL-3933.
+    // But so is ZERO WIDTH JOINER. See https://issues.bloomlibrary.org/youtrack/issue/BL-7081.
+    // The following variables are Control(format) characters that should split words.
+    // See list at: https://www.compart.com/en/unicode/category/Cf
+    var zeroWidthSplitters = "\u200b\u200c"; // ZERO WIDTH SPACE / ZERO WIDTH NON-JOINER
+    var ltrrtl = "\u200e\u200f"; // LEFT-TO-RIGHT MARK / RIGHT-TO-LEFT MARK
+    var directional = "\u202A-\u202E"; // more LTR/RTL/directional markers
+    var isolates = "\u2066-\u2069"; // directional "isolate" markers
+    regex = XRegExp(
```
Comment (hatton) [inline src/BloomBrowserUI/bookEdit/toolbox/readers/libSynphony/synphony_lib.js]: 1) It would help the readability of this if the \\p{Z} and \\p{Cc} were also assigned to variables. I *infer* that the old code had p{C} and that included more than we want but p{Cc} has less than we want. But for maintenance purposes, it would be easy to clarify that.
2) These should be const, not var.
